File: translation_output/w_test/gpt-4/codenet/java/python-sanitized/p03892.py
import sys
import math

def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

def solve(a, b, x, y):
    a = abs(x - a)
    b = abs(y - b)
    g = gcd(a, b)
    ans = (a // g) + (b // g) - 1
    return ans * g

def main():
    tokens = input().split()
    a, b, x, y = map(int, tokens)
    print(solve(a, b, x, y))

if __name__ == "__main__":
    main()

File: translation_output/no_test/codellama/codenet/python/java/p03892.java
<s> [INST] <<SYS>>
You are an expert java programmer and assistant
<</SYS>>

python code:

def gcd(a,b): return b if a%b==0 else gcd(b,a%b)
def main():
    a,b,c,d = map(int,input().split())
    h = abs(b-d)
    w = abs(c-a)
    if h==0 or w==0 : return 0
    m = gcd(max(h,w),min(h,w))
    return h+w-m

print(main())


Translate the above python code to java.

java code:

[/INST]

Here is the Java code equivalent to the Python code:
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int c = scanner.nextInt();
        int d = scanner.nextInt();
        int h = Math.abs(b - d);
        int w = Math.abs(c - a);
        if (h == 0 || w == 0) {
            return 0;
        }
        int m = gcd(Math.max(h, w), Math.min(h, w));
        return h + w - m;
    }

    public static int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
}
```
Note that the `gcd` function is defined as a separate method in the Java code, while it is defined as a lambda function in the Python code.</s>
